### DISCUSSION TITLE
When to use message vs subscribe callback method to receive published data ?

### DISCUSSION THREAD

**Discussion by luan-dang-techlabs** [mqttjs/MQTT.js#1785] [Q&A] [answered]
New to MQTT , trying to understand when you use 'message' callback to receive message vs 'subscribe' method to receive published method.

1. If I use both 'message' and 'subscribe' callback, does that mean i will receive in both places via published?

**Comment by robertsLando** [comment, marked as answer]
The subscribe callback is called with the subscription result, it tells you if there was an error on subscription or not. The `message` callback is called everytime a message is received.

Then there is also the `client.handleMessage(packet, done)` function that can be used to handle messages with backpressure support (the next message that comes is not processed until you call `done` callback.

**Reply by luan-dang-techlabs** [reply]
thanks - i see what you mean now. some reason i thought subscribe's callback has the payload from publish, but  ```message``` callback is used to receive payload.

**Reply by kiranpradeep** [reply]
I also thought same. Infact that would eased development and also helped readability. Now we have filter topics on '_message_' and then invoke appropriate handler.

Also if the client is used in multiple files, both files will receive same callbacks :(.
